File: STA/tblxinfo.m
function h=tblxinfo(tabl)
%
% h=tblxinfo(tabl) gives the transinformation, in bits
% tabl must be 2-dimensional.
%
% sum(tabl) assumed to be 1, and all elements positive.
%
% See also HISTINFO, HISTENT.
%
h=histinfo(sum(tabl,1))+histinfo(sum(tabl,2))-histinfo(tabl);
return

File: src/metricspace/entropy/tblxinfo.py
import numpy as np
from . import histinfo


def tblxinfo(tabl):
    """
    Calculates the transinformation, in bits.

    Args:
        tabl (numpy.ndarray): 2-dimensional array of probabilities.

    Returns:
        float: Transinformation in bits.
    """
    h = histinfo(np.sum(tabl, axis=1)) + histinfo(np.sum(tabl, axis=0)) - histinfo(tabl)
    return h
